New package: Yandex.YandexMessenger version 2.170.0

Dragon1573 opened the pull request:
## What's changed

- Resolve #119318

## Checks

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122155)

Dragon1573 committed:
New package: Yandex.Chats.YandexMessenger version 2.170.0

mdanish-kh commented on manifests/y/Yandex/Chats/YandexMessenger/2.170.0/Yandex.Chats.YandexMessenger.installer.yaml:
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Yandex.Chats.YandexMessenger
```
```suggestion
PackageIdentifier: Yandex.Messenger
```

PackageId should be kept concise where possible.

mdanish-kh commented on manifests/y/Yandex/Chats/YandexMessenger/2.170.0/Yandex.Chats.YandexMessenger.locale.ru.yaml:
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.5.0.schema.json
+
+PackageIdentifier: Yandex.Chats.YandexMessenger
```
```suggestion
PackageIdentifier: Yandex.Messenger
```

mdanish-kh commented on manifests/y/Yandex/Chats/YandexMessenger/2.170.0/Yandex.Chats.YandexMessenger.yaml:
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.5.0.schema.json
+
+PackageIdentifier: Yandex.Chats.YandexMessenger
```
```suggestion
PackageIdentifier: Yandex.Messenger
```

mdanish-kh commented on manifests/y/Yandex/Chats/YandexMessenger/2.170.0/Yandex.Chats.YandexMessenger.locale.ru.yaml:
```diff
@@ -0,0 +1,12 @@
+
+PackageIdentifier: Yandex.Chats.YandexMessenger
+PackageVersion: 2.170.0
+PackageLocale: ru
```
```suggestion
PackageLocale: en-US
```

PackageLocale should follow BCP-47 format. If this package is mainly used in Russia, then change the locale to `ru-RU` and provide ShortDescription in Russian. Otherwise we can make `en-US` as the default locale

mdanish-kh commented on manifests/y/Yandex/Chats/YandexMessenger/2.170.0/Yandex.Chats.YandexMessenger.locale.ru.yaml:
```diff
@@ -0,0 +1,12 @@
+PackageVersion: 2.170.0
+PackageLocale: ru
+Publisher: Yandex.Chats
+PackageName: Yandex Messenger 2.170.0
```
```suggestion
PackageName: Yandex Messenger
```

Fuzzy matching is used for PackageName. We can omit the version information that changes for every new release and it wont affect the package matching.

mdanish-kh reviewed (changes requested):
After committing the suggested changes, you would also have to modify the directory structure to match the new PackageIdentifier. `manifests/y/Yandex/Chats/YandexMessenger` should be renamed to `manifests/y/Yandex/Messenger`. Subsequently, the file names should also be changed to match the new Package ID.

A quick and easy way to do this would be here in GitHub Web by pressing the <kbd>.</kbd> (dot) key on your keyboard that'll open up the PR in VS Code Online. Navigate to Explorer menu, find the required files and make the appropriate changes.

After that, stage and commit your changes from Source Control menu on the left. (The changes are automatically pushed)

Dragon1573 committed:
Fixup: Change package identifier

Dragon1573 committed:
Fixup: Change package default locale

Dragon1573 committed:
Fixup: Change package name

- Fuzzy matching is used for PackageName. We can omit the version information that changes for every new release and it wont affect the package matching.

Dragon1573 committed:
Fixup: Package locale

- The "PackageVersion" in version file does not match the default locale file.

mdanish-kh commented on manifests/y/Yandex/YandexMessenger/2.170.0/Yandex.YandexMessenger.installer.yaml:
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Yandex.YandexMessenger
```
```suggestion
PackageIdentifier: Yandex.Messenger
```

mdanish-kh reviewed (changes requested):
Looking at other packages in https://github.com/microsoft/winget-pkgs/tree/master/manifests/y/Yandex, and also the fact that shorter PackageIds are preferred, I think we should really go with `Yandex.Messenger` instead of `Yandex.YandexMessenger`. The user already knows the messenger app is from `Yandex` from the publisher part in the PackageId. This should save a bit of typing for the end user

Dragon1573 committed:
Fixup: Change "PackageIdentifier"

- Also move the folder of manifests
